pkg/doc/signature/suite/jsonwebsignature2020/public_key_verifier.go

```diff
@@ -11,24 +11,45 @@ import (
 	"crypto/ecdsa"
 	"crypto/elliptic"
 	"errors"
+	"fmt"
 	"math/big"
 
+	"github.com/decred/dcrd/dcrec/secp256k1/v2"
+
 	sigverifier "github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
 )
 
-// PublicKeyVerifierP256 verifies a ECDSA signature taking public key bytes as input.
+// PublicKeyVerifierEC verifies a ECDSA signature taking public key bytes as input.
 // NOTE: this verifier is present for backward compatibility reasons and can be removed soon.
 // Please use CryptoVerifier or your own verifier.
-type PublicKeyVerifierP256 struct {
+type PublicKeyVerifierEC struct {
 }
 
 // Verify will verify a signature.
-func (v *PublicKeyVerifierP256) Verify(pubKey *sigverifier.PublicKey, doc, signature []byte) error {
-	pubKeyBytes := pubKey.Value
+func (v *PublicKeyVerifierEC) Verify(pubKey *sigverifier.PublicKey, doc, signature []byte) error {
+	var (
+		curve   elliptic.Curve
+		keySize int
+	)
+
+	switch pubKey.Curve {
+	case "P-256":
+		curve = elliptic.P256()
+		keySize = 32
+	case "P-384":
+		curve = elliptic.P384()
+		keySize = 48
+	case "P-521":
+		curve = elliptic.P521()
+		keySize = 66
+	case "secp256k1":
+		curve = secp256k1.S256()
+		keySize = 32
+	default:
+		return fmt.Errorf("ecdsa: unsupported elliptic curve '%s'", pubKey.Curve)
+	}
```

Comment:
this doesn't belong to Verify()

create a separate function

probably create a struct keyInfo that has a curve and keySize if needed in more than one place
probably expose it to be used for Sign as well ?

Reply:
OK! I will try to simplify it.